### ISSUE TITLE
[repl] Errors use bad code format

### ISSUE THREAD

**Issue by Brayan-724** [kaleidawave/ezno#210]
Using ezno from npm with deno.

![image](https://github.com/user-attachments/assets/9a5bb35a-5df4-40c3-8eed-c634fc954d19)

```
❯ nix run nixpkgs#deno -- run -A npm:ezno repl
Entering REPL. Exit with `close()`
> // Unused code
> const x: 5 = 4 + 2;
error: 
  ┌─ CLI.tsx:1:14
  │
1 │ // Unused codeconst x: 5 = 4 + 2;
  │          -   ^^^^^ Type 6 is not assignable to type 5
  │          │    
  │          Variable declared with type 5
```

It appears to don't save new lines properly, exploring a few I found that repl source code append happens [here](https://github.com/kaleidawave/ezno/blob/main/src/repl.rs#L80) and I think it could be fixed easy

**Comment by kaleidawave** [comment]
Interesting! Yes the REPL maintains a single file and appends entries to it. Haven't worked on the REPL in ages. I should check
1. That there is a new line character between entries
2. The "appended" AST has positions *offset* by the current source length

Will have a look at those two when I have the time. Maybe to do in #209

I wonder if there a way to add a test this? Need to spawn a REPL instance, pass some information over stdin and *expect* the stdout

---
```
1 │ // Unused codeconst x: 5 = 4 + 2;
  │          -   ^^^^^ Type 6 is not assignable to type 5
  │          │    
  │          Variable declared with type 5
```
*is being confused as*
```
1 │ const x: 5 = 4 + 2;
  │          -   ^^^^^ Type 6 is not assignable to type 5
  │          │    
  │          Variable declared with type 5
```

**Comment by kaleidawave** [comment]
Had a look and I think I can figure out a fix. I am also going to try and fix the other CLI JS problems #204 and #205 at the same time.

First I want to add a test. Moving the single file to use the new NodeJS testing framework. Also want to test the repl interaction. I don't know if @Brayan-724 you know about nodejs spawning a process, sending stdin, reading stdout. I am trying this but it hangs? [`925cec9` (#209)](https://github.com/kaleidawave/ezno/pull/209/commits/925cec9bd1cf03a7b845b84629c3b6f54b77fe5f#diff-bcec134af8d3d295a9a1a4055abe61c8b572f71d92cce58ead1a081f3fb22e33)

**Comment by kaleidawave** [comment]
This should be fixed in the latest release. Added some `.strip` calls to remove trailing new lines and stuff

![image](https://github.com/user-attachments/assets/3750b15d-8874-4906-88c3-c47df9c450cc)

> I think there is currently a parser bug with new lines that you may encounter. Other than that I have got it working now
